java/impl/aws/src/main/java/co/worklytics/psoxy/aws/AwsModule.java

```diff
@@ -86,20 +88,54 @@ static CognitoIdentityClient cognitoClient(AwsEnvironment awsEnvironment) {
     static ConfigService nativeConfigService(HostEnvironment hostEnvironment,
                                              EnvVarsConfigService envVarsConfigService,
                                              ParameterStoreConfigServiceFactory parameterStoreConfigServiceFactory,
-                                             @Named("instance") ParameterStoreConfigService instanceScopedConfigService) {
+                                             @Named("instance") ParameterStoreConfigService instanceScopedParameterConfigService,
+                                             SecretsManagerConfigServiceFactory secretsManagerConfigServiceFactory,
+                                             @Named("instance") SecretsManagerConfigService instanceScopedSecretsManagerConfigService) {
 
         String pathToSharedConfig =
                 envVarsConfigService.getConfigPropertyAsOptional(ProxyConfigProperty.PATH_TO_SHARED_CONFIG)
                         .orElse(null);
 
-        ParameterStoreConfigService sharedConfigService =
+        ParameterStoreConfigService sharedParameterConfigService =
                 parameterStoreConfigServiceFactory.create(pathToSharedConfig);
 
+        AwsEnvironment.SecretStoreImplementations secretStoreImpl = envVarsConfigService.getConfigPropertyAsOptional(AwsEnvironment.AwsConfigProperty.SECRETS_STORE)
+            .map(String::toUpperCase) //case-insensitive, so accept 'aws_ssm_parameter_store'
+            .map(AwsEnvironment.SecretStoreImplementations::valueOf)
+            .orElse(AwsEnvironment.SecretStoreImplementations.AWS_SSM_PARAMETER_STORE);
+
+        ConfigService sharedConfigService;
+        ConfigService instanceConfigService;
+        if (secretStoreImpl == AwsEnvironment.SecretStoreImplementations.AWS_SECRETS_MANAGER) {
+            instanceConfigService = CompositeConfigService.builder()
+                .preferred(instanceScopedSecretsManagerConfigService)
+                .fallback(instanceScopedParameterConfigService)
```

Comment:
I don't think we should fallback to SSM. If the customer has specified the secrets manager we should stick to that. Will that even work from an infra perspective? If for whatever reason on infra side this would work could give the false sense that everything is under SM but is not.
Is this to support migrations from SSM to SM?

Reply:
our problem there is config values that AREN'T secrets.  Those are easier to inspect and change via AWS console than Secrets Manager secret.  

Although in the end for simplicity I left the best case of this (client id) as a secret just in case.

Reply:
and actually, also problem for `RULES` and `ADDITIONAL_TRANSFORMS`; those don't fit as env vars.

what we need is to add the concept of whether a config value is 'secret' or not; and then pull secrets from the "secret store" and non-secret from regular config service (for which env vars falling through to SSM is more practical)

that said, env vars are legitimate for secrets as well, tbh.  just hard to manage as AWS ux renders as ~80 char html input.  YAML/JSON/etc that's multi-line is unworkable. And unlike SSM SecureString, values shown unmasked in the UX.